interrupts.c

```diff
@@ -99,15 +101,35 @@ static const char *exceptions[32] = {
   [EXC_MCHECK] = "Machine checkcore",
 };
 
+#define PDE_ID_FROM_PTE_ADDR(x) (((x) & 0x003ff000) >> 12)
+
 __attribute__((interrupt)) 
 void tlb_exception_handler()
 {
   int code = (mips32_get_c0(C0_CAUSE) & CR_X_MASK) >> CR_X_SHIFT;
-
+  uint32_t vaddr = mips32_get_c0(C0_BADVADDR);
   kprintf("[tlb] %s at $%08x!\n", exceptions[code], mips32_get_c0(C0_ERRPC));
-  kprintf("[tlb] Caused by reference to $%08x!\n", mips32_get_c0(C0_BADVADDR));
-
-  panic("[tlb] TLB exception handler unimplemented\n");
+  kprintf("[tlb] Caused by reference to $%08x!\n", vaddr);
```

Comment:
Sure, `log` is right tool for the job here. Efficient and flexible logging framework is another story - don't be too much concerned about it for now.